.github/workflows/build.yml

```diff
@@ -124,58 +128,21 @@ jobs:
         with:
           vs-version: '[17.9,)'
 
-      - name: Enable ${{ env.TARGET_PLATFORMS }} TargetFrameworks
-        working-directory: ./${{ env.MULTI_TARGET_DIRECTORY }}
-        run: powershell -version 5.1 -command "./UseTargetFrameworks.ps1 ${{ env.TARGET_PLATFORMS }}" -ErrorAction Stop
-
-      - name: Generate solution w/ ${{ env.TEST_PLATFORM }} Tests
+      # Generate full solution with all projects (sample gallery heads, components, tests)
+      - name: Generate solution with ${{ matrix.multitarget }} gallery, components and tests
         working-directory: ./
-        run: powershell -version 5.1 -command "./tooling/GenerateAllSolution.ps1 -IncludeHeads ${{ env.TEST_PLATFORM }}${{ env.ENABLE_DIAGNOSTICS == 'true' && ' -UseDiagnostics' || '' }}" -ErrorAction Stop
-
-      - name: Enable Uno.WinUI (in WinUI3 matrix only)
-        if: ${{ matrix.platform == 'WinUI3' }}
-        working-directory: ./${{ env.MULTI_TARGET_DIRECTORY }}
-        run: powershell -version 5.1 -command "./UseUnoWinUI.ps1 3" -ErrorAction Stop
-
-      - name: Format Date/Time of Commit for Package Version
-        if: ${{ env.IS_RELEASE == 'false' }}
-        run: |
-          echo "VERSION_DATE=$(git log -1 --format=%cd --date=format:%y%m%d)" >> $env:GITHUB_ENV
-
-      # Semver regex: https://regex101.com/r/Ly7O1x/3/
-      - name: Format Date/Time of Release Package Version
-        if: ${{ env.IS_RELEASE == 'true' }}
-        run: |
-          $ref = "${{ github.ref }}"
-          $ref -match "^refs/heads/rel/(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:-(?<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
-          echo "VERSION_DATE=$($matches['patch'])" >> $env:GITHUB_ENV
-          echo "VERSION_PROPERTY=$($matches['prerelease'])" >> $env:GITHUB_ENV
+        run: powershell -version 5.1 -command "./tooling/GenerateAllSolution.ps1 -MultiTargets ${{ matrix.multitarget }} ${{ env.ENABLE_DIAGNOSTICS == 'true' && ' -UseDiagnostics' || '' }} -WinUIMajorVersion ${{ matrix.winui }}" -ErrorAction Stop
```

Comment:
This does the use target and uno switch now all together, eh?

Reply:
Yep, since this is the entrypoint that most people and our CI uses for opening the toolkit solution, I've opted to forward this option via parameter rather than having the consumer manually invoke a separate script.